### PR TITLE
Add weekday to time contidion editor 

### LINKED ISSUE #3833: Automations condition.weekday missing in Automation GUI
The weekday interface in the Automations UI is missing. 
I have the following config:
```yml
- id: '1555796295451'
  alias: Verwarming uit MA DI WO DO
  trigger:
  - at: '11:30'
    platform: time
  condition:   
  - condition: time
    weekday:
    - mon
    - tue
    - wed
    - thu
  action:
  - data:
      entity_id: climate.woonkamer,climate.serre
      temperature: 16
    service: climate.set_temperature
  - data:
      entity_id: automation.schuifpui_dicht_zet_verwarming_aan
    service: automation.turn_off
```
But the interface looks like this
image
It would be nice to have a <select> or something to select the weekday time condition and then a list of checkboxes/switches to add or remove the weekdays that will be added in the condition.

### PR DESCRIPTION
**Pull request by Misiu**
## Proposed change

**Replacement of https://github.com/home-assistant/frontend/pull/6786 (I messed up rebase, it is a small change, so it was easier to create new PR)**
image

This is the first step to add weekday's support to time condition.
Next will be `ha-expand-panel` (https://github.com/home-assistant/frontend/issues/6726) that will allow us to hide (collapse) parts that aren't filled

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #3833
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by Misiu** [comment]
@zsarnett could you take a look at this one?
It's a proper rebase of https://github.com/home-assistant/frontend/pull/6786 with all the comments and suggestions applied.

**Review by zsarnett — APPROVED** [review]

**Comment by zsarnett** [comment]
I'll let Bram take the final look before merging. Also, I have only reviewed the code and not the actual implementation and usability. I may look at that later today

**Comment by Misiu** [comment]
@zsarnett thank you for that.
I've checked the code before I created the initial PR, but core developer opinion is much appreciated.
Any chance this might get added to 0.115?

**Comment by zsarnett** [comment]
No unfortunately this will not be in 0.115. Beta Cut was yesterday. Bug fixes and follow up PRs only for this release

**Comment by Misiu** [comment]
@bramkragten could you take a look at this? I already got @zsarnett approval, so we're one step closer to finally fixing/adding this functionality.
It was asked in WTH: https://community.home-assistant.io/t/wth-day-of-the-week-in-automations/226902

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts
Comment (bramkragten) [inline src/panels/config/automation/condition/types/ha-automation-condition-time.ts:204]:
```suggestion
      value: {...this.condition, weekday},
```

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts
Comment (bramkragten) [inline src/panels/config/automation/condition/types/ha-automation-condition-time.ts:201]:
```suggestion
```

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts
Comment (bramkragten) [inline src/panels/config/automation/condition/types/ha-automation-condition-time.ts:212]: Why do we set the height to 40px? What is wrong with the default styling?

Comment (Misiu) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts:212]: It looks really bad without those 40px:

1403c6bcebf56314e31d3a1dea595480

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts
Comment (Misiu) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts]: done

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts
Comment (Misiu) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts]: done

**Comment by bramkragten** [comment]
If you don't set any day, it will trigger at every day, right? That is not clear from the UI

**Comment by Misiu** [comment]
> If you don't set any day, it will trigger at every day, right? That is not clear from the UI

Yes, that's right and I agree this isn't clear. Ideally, I should require at least one selected day and show warning/block saving if none are selected.
This PR is the first part. In the second part, I want to put the time part in one expansion panel and date in the second panel, so there will be much more space to add info.

What do you think about adding a title to the section and add a tooltip to it, something like here: https://github.com/home-assistant/frontend/pull/7006#issuecomment-696462796

**Comment by bramkragten** [comment]
Why not just turn the logic around and enable all switches by default? That would reflect the actual behavior without the need for explaining?

**Comment by Misiu** [comment]
@bramkragten I can do that, but how to inform the user that he will get the same behavior if he selects all or none?

**Comment by bramkragten** [comment]
Saving it with none selected makes no sense, then it would not be true ever. So we should not allow that.

**Comment by Misiu** [comment]
> Btw, an empty array will not result in the same behavior as all selected. Only having the weekday not set at all will.
> https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/condition.py#L526-L534

If I understand that correctly if someone deselects all days then the condition will always be False? So right, that shouldn't be allowed.

> Saving it with none selected makes no sense, then it would not be true ever. So we should not allow that.

Is there a way to prevent saving automation/mark condition as not valid? I can show an error message (custom logic) but maybe there is something I can reuse.

**Comment by bramkragten** [comment]
Well, we don't have to? As it would be logical that if you don't select any days it will not match ever?

**Comment by Misiu** [comment]
@bramkragten ok, I'll turn the logic around, and in future PR (with expansion panels) I'll add a tooltip with info that disabling all days will always make condition False.

**Comment by Misiu** [comment]
@bramkragten I've changed:
```ts
@property({ attribute: false }) public condition!: TimeCondition
```
to 
```ts
  private _condition!: TimeCondition;

  set condition(value:TimeCondition) {
    this._condition = value;
    this._condition.weekday = this._condition.weekday || ["mon","tue", "wed","thu","fri", "sat","sun"];
  }

  get condition() {
    return this._condition;
  }
```

Is that fine? I want to assign all days if condition.weekday is empty or null.

**Comment by bramkragten** [comment]
Just set the switches to checked when `!condition.weekday` and when any is changed and `!condition.weekday`, `condition.weekday = ["mon","tue", "wed","thu","fri", "sat","sun"]` then do the same as before.

**Comment by Misiu** [comment]
@bramkragten please take a final look at this.
Now when there is no `weekday` in condition all days are displayed as selected. If someone doesn't change any day then the condition is saved without the `weekday` part (just like it used to do before), but if you deselect one day then the `weekday` part is generated and saved.

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts
Comment (bramkragten) [inline src/panels/config/automation/condition/types/ha-automation-condition-time.ts:163]:
```suggestion
              .checked=${!weekday || weekday.includes(day)}
```

Comment (bramkragten) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts:163]: weekday can btw also be a string, so that should also be checked, and the type adjusted

Comment (Misiu) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts:163]: done

Comment (Misiu) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts]: > weekday can btw also be a string, so that should also be checked, and the type adjusted

I've added string[] as a type: https://github.com/home-assistant/frontend/pull/6848/files#diff-be629ff01fe2971d4cb277c22d528d74R173

File: src/data/automation.ts
Comment (bramkragten) [inline src/data/automation.ts:173]:
```suggestion
  weekday?: string | string[];
```

File: src/data/automation.ts
Comment (bramkragten) [inline src/data/automation.ts:172]:
```suggestion
  after?: string;
  before?: string;
```

File: src/data/automation.ts
Comment (Misiu) [reply src/data/automation.ts:173]: I'm changing that right now

Comment (Misiu) [reply src/data/automation.ts]: done

File: src/data/automation.ts
Comment (Misiu) [reply src/data/automation.ts]: done

**Review by bramkragten — APPROVED** [review]

**Comment by Misiu** [comment]
@bramkragten @zsarnett thank you guys for the reviews!

**Pull request merged**
